app/api/helpers/tasks.py

```diff
@@ -51,56 +51,66 @@
+                    f.close()
+                encoded = base64.b64encode(file_data).decode()
+                attachment = Attachment()
+                attachment.file_content = FileContent(encoded)
+                attachment.file_type = FileType('application/pdf')
+                attachment.file_name = FileName(payload['to'])
+                attachment.disposition = Disposition('attachment')
+                message.add_attachment(attachment)
+        sendgrid_client = SendGridAPIClient(get_settings()['sendgrid_key'])
+        logging.info('Sending an email regarding {} on behalf of {}'.format(payload["subject"], payload["from"]))
         sendgrid_client.send(message)
         logging.info('Email sent successfully')
+    except requests.exceptions.HTTPError as e:
+        if e.code == 429:
+            logging.exception("Sendgrid quota has exceeded")
+        else:
+            logging.exception("The following error has occurred with sendgrid-{}".format(str(e)))
     except Exception:
-        logging.exception('Error occured while sending the email')
+        logging.exception('Error in sending mail via Sendgrid')
 
 
 @celery.task(name='send.email.post.smtp')
-def send_mail_via_smtp_task(config, payload):
-    mailer_config = {
-        'transport': {
-            'use': 'smtp',
-            'host': config['host'],
-            'username': config['username'],
-            'password': config['password'],
-            'tls': config['encryption'],
-            'port': config['port']
-        }
-    }
-
-    mailer = Mailer(mailer_config)
-    mailer.start()
-    message = Message(author=payload['from'], to=payload['to'])
-    message.subject = payload['subject']
-    message.plain = strip_tags(payload['html'])
-    message.rich = payload['html']
-    message.attach(name=payload['attachments'])
-    mailer.send(message)
-    logging.info('Message sent via SMTP')
-    mailer.stop()
+def send_email_task_smtp(payload, config, headers=None):
+    try:
+        mailer_config = {
+                'transport': {
+                    'use': 'smtp',
+                    'host': config['host'],
+                    'username': config['username'],
+                    'password': config['password'],
+                    'tls': config['encryption'],
+                    'port': config['port']
+                }
+            }
+
+        mailer = Mailer(mailer_config)
+        mailer.start()
+        message = Message(author=payload['from'], to=payload['to'])
+        message.subject = payload['subject']
+        message.plain = strip_tags(payload['html'])
+        message.rich = payload['html']
+        if payload['attachments'] is not None:
+            for attachment in payload['attachments']:
+                message.attach(name=attachment)
+        mailer.send(message)
+        logging.info('Message sent via SMTP')
+        mailer.stop()
+    except Exception:
+        logging.exception('Error in sending mail via SMTP')
```

Comment:
Unless you do something in the block, generic exception catching is useless

Reply:
@iamareebjamal Will catching a specific exception and logging the error message do? Like `DeliveryFailedException` & `MessageFailedException`?

Reply:
Why do you need to catch that. Let it be thrown